### PR TITLE
Project - Framework Structure Tweaks

### PR DESCRIPTION
**Pull request by cnoon**
Dug through the new 3.0.0 structure and found a few things that could be condensed or unified. I tried to split each change out into a separate commit to make it easier to see what changes I actually made. 

The only callout I have is that I changed the deployment target in the project on iOS to 8.0 to help the compiler out to generate non-compatible changes. However, the podspec is still supporting iOS 7.0 as a deployment target. Does CocoaPods handle the difference between 7 and 8 support for frameworks properly? I'm assuming it does, but wanted to ask. I've spent so much time in Swift over the last year, I'm not even sure how this works for Obj-C libraries.

### CODE REVIEW

**Comment by cnoon** [comment]
cc @kcharwood and @kylef

**Comment by cnoon** [comment]
@kcharwood I also changed the release version from `3.0.0-beta0` to `3.0.0-beta.1` to follow the Alamofire convention. If you'd prefer the syntax you used, then that's perfectly fine too. Just though I'd throw in a commit to unify our approach if you wanted.

File: AFNetworking.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name     = 'AFNetworking'
-  s.version  = '3.0.0'
+  s.version  = '3.0.0-beta.1'
```
Comment (kylef) [inline AFNetworking.podspec]: I'm confused with this change, isn't 3.0.0 already out? https://github.com/CocoaPods/Specs/blob/master/Specs/Alamofire/3.0.0/Alamofire.podspec.json

Comment (kylef) [inline AFNetworking.podspec]: Oops sorry, too late in the day. Mistaken AFNetworking for Alamofire.

Comment (cnoon) [inline AFNetworking.podspec]: Sleep my man...sleep! 💤😉

**Comment by tomtaylor** [comment]
I'm having trouble building using Carthage on 122f1b2965b5781f3764796795de575bde9ea813:

```
*** Building scheme "AFNetworking tvOS" in AFNetworking.xcworkspace
xcodebuild: error: Failed to build workspace AFNetworking with scheme AFNetworking tvOS.
	Reason: The run destination My Mac is not valid for Running the scheme 'AFNetworking tvOS'.
A shell task failed with exit code 70:
xcodebuild: error: Failed to build workspace AFNetworking with scheme AFNetworking tvOS.
	Reason: The run destination My Mac is not valid for Running the scheme 'AFNetworking tvOS'.
```

**Comment by tomtaylor** [comment]
Ignore this - I've realised that it's trying to build the tvOS target, but the SDK isn't provided on Xcode 7.0. Building with `--platform iOS` bypasses this.

**Comment by kcharwood** [comment]
Thanks @cnoon. These all look good to me! :beers:

I'm also cool with the version name change.

**Pull request merged**